File: Golang/744. Find Smallest Letter Greater Than Target.go
// func nextGreatestLetter(letters []byte, target byte) byte { // 4ms
//     min := byte('z')
//     for _, letter := range letters {
//         if letter > target {
//             return letter
//         }
//         if letter < min {
//             min = letter
//         }
//     }
//     return min
// }

func nextGreatestLetter(letters []byte, target byte) byte { // 4ms
    min := rune('z')
    for _, letter := range letters {
        if rune(letter) > rune(target) {
            return letter
        }
        if rune(letter) < min {
            min = rune(letter)
        }
    }
    return byte(min)
}

File: Python3/744. Find Smallest Letter Greater Than Target.py
class Solution:
    def nextGreatestLetter(self, letters, target):
        """
        :type letters: List[str]
        :type target: str
        :rtype: str
        """
        charDict = dict(zip(string.ascii_letters[:26], range(26)))
        nums = [charDict[i] for i in letters]
        target = charDict[target]
        for num in sorted(nums):
            if num > target:
                return string.ascii_letters[num]
        return string.ascii_letters[nums[0]]
